core/services/synchronization/telemetry_ingress_batch_client.go

```diff
@@ -0,0 +1,301 @@
+package synchronization
+
+import (
+	"context"
+	"errors"
+	"net/url"
+	"sync"
+	"time"
+
+	"go.uber.org/atomic"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services"
+	"github.com/smartcontractkit/chainlink/core/services/keystore"
+	telemPb "github.com/smartcontractkit/chainlink/core/services/synchronization/telem"
+	"github.com/smartcontractkit/chainlink/core/utils"
+	"github.com/smartcontractkit/wsrpc"
+	"github.com/smartcontractkit/wsrpc/examples/simple/keys"
+)
+
+//go:generate mockery --dir ./telem --name TelemClient --output ./mocks/ --case=underscore
+
+// TelemetryIngressBatchClient encapsulates all the functionality needed to
+// send telemetry to the ingress server using wsrpc
+type TelemetryIngressBatchClient interface {
+	services.Service
+	Start() error
+	Close() error
+	Send(TelemPayload)
+}
+
+// NoopTelemetryIngressBatchClient is a no-op interface for TelemetryIngressBatchClient
+type NoopTelemetryIngressBatchClient struct{}
+
+// Start is a no-op
+func (NoopTelemetryIngressBatchClient) Start() error { return nil }
+
+// Close is a no-op
+func (NoopTelemetryIngressBatchClient) Close() error { return nil }
+
+// Send is a no-op
+func (NoopTelemetryIngressBatchClient) Send(TelemPayload) {}
+
+// Healthy is a no-op
+func (NoopTelemetryIngressBatchClient) Healthy() error { return nil }
+
+// Ready is a no-op
+func (NoopTelemetryIngressBatchClient) Ready() error { return nil }
+
+type telemetryIngressBatchClient struct {
+	utils.StartStopOnce
+	url             *url.URL
+	ks              keystore.CSA
+	serverPubKeyHex string
+
+	telemClient  telemPb.TelemClient
+	globalLogger logger.Logger
+	logging      bool
+	lggr         logger.Logger
+
+	wgDone sync.WaitGroup
+	chDone chan struct{}
+
+	telemBufferSize   uint
+	telemMaxBatchSize uint
+	telemSendInterval time.Duration
+
+	workers      map[string]*telemetryIngressBatchWorker
+	workersMutex sync.Mutex
+}
+
+// NewTelemetryIngressBatchClient returns a client backed by wsrpc that
+// can send telemetry to the telemetry ingress server
+func NewTelemetryIngressBatchClient(url *url.URL, serverPubKeyHex string, ks keystore.CSA, logging bool, lggr logger.Logger, telemBufferSize uint, telemMaxBatchSize uint, telemSendInterval time.Duration) TelemetryIngressBatchClient {
+	return &telemetryIngressBatchClient{
+		telemBufferSize:   telemBufferSize,
+		telemMaxBatchSize: telemMaxBatchSize,
+		telemSendInterval: telemSendInterval,
+		url:               url,
+		ks:                ks,
+		serverPubKeyHex:   serverPubKeyHex,
+		globalLogger:      lggr,
+		logging:           logging,
+		lggr:              lggr.Named("TelemetryIngressBatchClient"),
```

Comment:
We should have all names for all named loggers in one place here: https://github.com/smartcontractkit/chainlink/blob/567932f0bc793a5ca1804a6dfa40863793748769/core/logger/logger.go#L100

Reply:
Just to confirm, I should add `TelemetryIngressBatchClient` and `TelemetryIngressBatchWorker` to this list of constants? I see there's a bunch of named loggers throughout the app not listed there

Reply:
We are trying to make everything looking identical, and at least every new code should do that.